src/game.cpp

```diff
@@ -10,7 +10,97 @@ void Game::generateNewPiece() {
 	this->currPiece = pieceTypes[rand() % PIECES::NUMBER_OF_PIECES](this->board);
 }
 
+void Game::init() {
+	this->window.create(sf::VideoMode(200, 400), "Tetris");
+	generateNewPiece();
+}
+
+void Game::update() {
+	while (this->window.isOpen()) {
+		handleEvents();
+		handlePiecePassiveMoveDown();
+	}
+}
+
+void Game::terminate() {
+	this->window.close();
+}
+
+void Game::handleEvents() {
+	sf::Event event;
+	while (this->window.pollEvent(event)) {
+		switch (event.type) {
+			case sf::Event::Closed:
+				terminate();
+			case sf::Event::KeyPressed:
+				switch (event.key.code) {
+					case sf::Keyboard::Down:
+						handlePieceMoveDown();
+					case sf::Keyboard::Left:
+						handlePieceMoveLeft();
+					case sf::Keyboard::Right:
+						handlePieceMoveLeft();
+					case sf::Keyboard::Up:
+						handlePieceRotate();
+					case sf::Keyboard::Space:
+						handlePieceDrop();
+				}
+		}
+	}
+}
+
+void Game::handlePiecePassiveMoveDown() {
+	if (this->passiveMoveDownTimer.getElapsedTime().asSeconds() > 1.f) {
+		if (!this->currPiece->moveDown()) {
+			this->currPiece->set();
+			generateNewPiece();
+		}
+		this->passiveMoveDownTimer.restart();
+	}
+}
+
+void Game::handlePieceMoveDown() {
+	if (this->movePieceDownTimer.getElapsedTime().asSeconds() > .2f) {
+		if (!this->currPiece->moveDown()) {
+			this->currPiece->set();
+			generateNewPiece();
+		}
+		this->movePieceDownTimer.restart();
+		this->passiveMoveDownTimer.restart();
+	}
+}
+
+void Game::handlePieceMoveLeft() {
+	if (this->movePieceLeftTimer.getElapsedTime().asSeconds() > .2f) {
+		this->currPiece->moveLeft();
+		this->movePieceLeftTimer.restart();
+	}
+}
+
+void Game::handlePieceMoveRight() {
+	if (this->movePieceLeftTimer.getElapsedTime().asSeconds() > .2f) {
+		this->currPiece->moveLeft();
+		this->movePieceLeftTimer.restart();
+	}
+}
+
+void Game::handlePieceDrop() {
+	this->currPiece->drop();
+	this->currPiece->set();
+	this->passiveMoveDownTimer.restart();
+}
+
+void Game::handlePieceRotate() {
+	this->currPiece->rotate();
+	this->passiveMoveDownTimer.restart();
```

Comment:
so you restart the passive move down timer everytime the user rotates a piece. So if a user keeps rotating the piece won't go down? That doesnt seem right afaik